docs/pages/docs/guides/custom-admin-ui-navigation.mdx

```diff
@@ -0,0 +1,214 @@
+import { Markdown } from '../../../components/Markdown';
+
+# Custom Admin UI Navigation
+
+In this guide, we'll illustrate how to create a custom Navigation component with custom routes to be rendered in the Admin-UI.
+
+## Creating your custom Navigation component
+
+### Navigation props
+
+Keystone passes the following props to your custom Navigation component:
+
+```ts
+type NavigationProps {
+    lists: ListMeta[];
+    authenticatedItem: AuthenticatedItem;
+}
+```
+
+`lists` is an array of list objects that include amongst other things routing information to be rendered by the custom Navigation component.
+
+```ts
+type ListMeta = {
+  // key is used as the unique key for each navigation item in React.
+  key: string;
+  // href to the list route in the Admin-UI.
+  path: string;
+  // Used as the label for each navigation item.
+  label: string;
+  /** There are other properties as well */
+};
+
+```
+
+`authenticatedItem` is a union of potential authentication states, expanded on below:
+
+```ts
+type AuthenticatedItem =
+  | { state: 'unauthenticated' }
+  | { state: 'authenticated'; label: string; id: string; listKey: string }
+  | { state: 'loading' }
+  | { state: 'error'; error: Error | readonly [GraphQLError, ...GraphQLError[]] };
+```
+
+This should be handled for you automatically by passing the prop into the `NavigationContainer`,
+
+!> `NavigationContainer` is one of many assistive components to help you more easily compose your custom Navigation component. You can read more about it down [below](#components).
+
+?> If you wish to render out the `authenticatedItem` in a custom manner, please be aware that you will need to handle all of the possible states of the `authenticatedItem`. Explaining how to do this is out of the scope of this particular guide.
+
+### Components
```

Comment:
Honestly this is definitely not the right place for *this* kind of documentation. 

Very much taking a 'let's get this content in first and figure out where it goes later' approach here.